spec/models/schema_contract/validator_spec.rb

```diff
@@ -0,0 +1,192 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require_relative Rails.root.join('app', 'models', 'schema_contract', 'validator')
+
+describe SchemaContract::Validator, aggregate_failures: true do
+  describe '#validate' do
+    let(:fixture) { 'spec/fixtures/schema_contract/test_schema.json' }
+    let(:test_data) { Rails.root.join(fixture).read }
+    let(:contract_record) do
+      create(:schema_contract_validation, response:)
+    end
+    let(:matching_response) do
+      {
+        data: [
+          {
+            required_string: '1234',
+            required_object: {
+              required_nested_string: 'required'
+            }
+          }
+        ],
+        meta: [{}]
+      }
+    end
+    let(:uuid_regex) { /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/ }
+
+    context 'when response matches schema' do
+      let(:response) { matching_response }
+
+      it 'updates record and does not log errors' do
+        expect do
+          SchemaContract::Validator.new(contract_record.id).validate
+        end.not_to raise_error(SchemaContract::Validator::SchemaContractValidationError)
+        expect(contract_record.reload.status).to eq('success')
+      end
+    end
+
+    ###
+    # the following tests are duplicative of the underlying JSON::Validator gem specs and are non-exhaustive
+    # but are useful for validating that this works as expected and for documenting expected behavior
+    ###
+
```

Comment:
Per the comment, I think these tests are somewhat duplicative, but I wanted to be sure that things really worked and also provide some examples to anyone writing contract schemas. We've often failed to use `additionalProperties: false` which is one reason our schema tests are not very useful.